yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,439 @@
+    ///         unwrapping extra rewards. This call is only used during normal
+    ///         operation of a strategy, and should be optimized to minimize
+    ///         losses as much as possible. This strategy implements the
+    ///         aforementioned behavior by getting KEEP tokens from the Saddle
+    ///         reward pool, using obtained tokens to buy one of the Saddle
+    ///         pool's accepted token and depositing that token back to the
+    ///         Saddle pool. This way the strategy is gaining new vault's
+    ///         underlying tokens thus making the profit.
+    /// @param debtOutstanding Will be 0 if the strategy is not past the
+    ///        configured debt limit, otherwise its value will be how far past
+    ///        the debt limit the strategy is. The strategy's debt limit is
+    ///        configured in the vault.
+    /// @return profit Amount of realized profit.
+    /// @return loss Amount of realized loss.
+    /// @return debtPayment Amount of repaid debt. The value of debtPayment
+    ///         should be less than or equal to debtOutstanding. It is okay for
+    ///         it to be less than debtOutstanding, as that should only used as
+    ///         a guide for how much is left to pay back. Payments should be
+    ///         made to minimize loss from slippage, debt, withdrawal fees, etc.
+    function prepareReturn(uint256 debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 profit,
+            uint256 loss,
+            uint256 debtPayment
+        )
+    {
+        // Get the initial balance of the vault's underlying token under
+        // strategy management.
+        uint256 initialWantBalance = balanceOfWant();
+
+        // Get KEEP rewards from the Saddle reward pool.
+        ILPRewards(tbtcSaddleLPRewards).getReward();
+
+        // Buy WBTC using obtained KEEP tokens.
+        uint256 keepBalance = IERC20(keepToken).balanceOf(address(this));
+        if (keepBalance > 0) {
+            IERC20(keepToken).safeIncreaseAllowance(uniswap, keepBalance);
+
+            address[] memory path = new address[](3);
+            path[0] = keepToken;
+            path[1] = wethToken;
+            path[2] = wbtcToken;
+
+            IUniswapV2Router(uniswap).swapExactTokensForTokens(
+                keepBalance,
+                0,
+                path,
+                address(this),
+                now
+            );
+        }
+
+        // Deposit acquired WBTC to the Saddle pool to gain additional
+        // vault's underlying tokens.
+        uint256 wbtcBalance = IERC20(wbtcToken).balanceOf(address(this));
+        if (wbtcBalance > 0) {
+            IERC20(wbtcToken).safeIncreaseAllowance(
+                tbtcSaddlePoolSwap,
+                wbtcBalance
+            );
+
+            uint256[] memory amounts = new uint256[](4);
+            amounts[1] = wbtcBalance;
```

Comment:
Can we add a TODO here, just like we have in other contracts and in `ethToWant` methods:
```
// TODO: When the new curve pool with tBTC v2 is deployed, verify that
//       the index of wBTC in the array is correct.
```
?

Reply:
Sure, done in https://github.com/keep-network/tbtc-v2/pull/13/commits/04cf745a5c63b0cdffc59de7c4a839b82191d66f.